### PR TITLE
add option to not relink

### PR DESCRIPTION
**Pull request by dberenbaum**
Adds `dvc add --no-relink` and `dvc commit --no-relink` as a way to speed up add, but at the potential cost of lots of copies and maybe having to relink later.

Related to:
- #9813
- #7607
- https://github.com/iterative/dvc-data/issues/274

### CODE REVIEW

File: dvc/commands/add.py [hunk 1]
```diff
@@ -119,6 +120,13 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Override local file or folder if exists.",
     )
+    parser.add_argument(
+        "--no-relink",
+        dest="relink",
+        action="store_false",
+        help="Don't recreate links from cache to workspace.",
+    )
```
Comment (skshetry) [inline dvc/commands/add.py:128]: I'd avoid introducing this asymmetry here. 

(What you have done is useful if you have support for both negative and positive behavors. Eg: you'd see this in `git`: `git stash --include-untracked` and `git stash --no-include-untracked`).

```suggestion
    parser.add_argument(
        "--no-relink",
        default=False,
        action="store_true",
        help="Don't recreate links from cache to workspace.",
    )
```

File: dvc/commands/add.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ def run(self):
                 to_remote=self.args.to_remote,
                 remote_jobs=self.args.remote_jobs,
                 force=self.args.force,
+                relink=self.args.relink,
```
Comment (skshetry) [inline dvc/commands/add.py:54]: Based on https://github.com/iterative/dvc/pull/10389/files#r1565145737.

```suggestion
                relink=not self.args.no_relink,
```

File: dvc/commands/add.py (hunk 1 above)
Comment (skshetry) [reply dvc/commands/add.py:128]: Same in `commit`.

**Review by skshetry — APPROVED** [review]

File: dvc/commands/add.py (hunk 1 above)
Comment (dberenbaum) [reply dvc/commands/add.py:128]: @skshetry Should we add both? We already have `dvc checkout --relink`. We could have `--relink` and `--no-relink` options for all of those commands even if it's redundant when it's the default.

Comment (skshetry) [reply dvc/commands/add.py:128]: We haven't done anything like that before in dvc. So maybe better to not introduce them. But up to you.

Comment (dberenbaum) [reply dvc/commands/add.py:128]: I'm seeing plenty of examples of either `store_false` and `store_true` in the codebase for similar situations, and I think it is just as awkward to use `store_true` (I will have to do `relink=not self.args.no_relink`), so I'm going to leave this as is.

**Pull request merged**